### PR TITLE
Draft: Log message improvements

### PR DESCRIPTION
**Pull request by albertony**

### CODE REVIEW

File: src/lib/net/SecureSocket.cpp [hunk 1]
```diff
@@ -695,37 +700,34 @@ SecureSocket::verifyCertFingerprint()
...
     LOG((CLOG_NOTE "server fingerprint: %s", fingerprint.c_str()));
 
-    std::string trustedServersFilename;
-    trustedServersFilename = barrier::string::sprintf(
-        "%s/%s/%s",
-        DataDirectories::profile().c_str(),
-        kFingerprintDirName,
-        kFingerprintTrustedServersFilename);
+    std::string trustedServersFilename = PathUtilities::concat(DataDirectories::profile(),
+        barrier::string::sprintf("%s%c%s",
```
Comment (p12tic) [inline src/lib/net/SecureSocket.cpp:704]: I think we should use PathUtilities::concat here too. Is there any reason not to do that?

File: src/lib/common/PathUtilities.h [hunk 2]
```diff
@@ -25,6 +25,7 @@ class PathUtilities
     static std::string basename(const std::string& path);
     static std::string concat(const std::string& left, const std::string& right);
 
+    static const char DefaultDelimiter;
```
Comment (p12tic) [inline src/lib/common/PathUtilities.h:28]: This can be removed when the single location where it's used starts using the `concat` function.

**Review by p12tic — CHANGES_REQUESTED** [review]
Just a couple of comments, otherwise good. It would be great if the path changes and log changes are split into separate commits.

File: src/lib/net/SecureSocket.cpp (hunk 1 above)
Comment (albertony) [reply src/lib/net/SecureSocket.cpp:704]: Just that it is not necessary to perform the searching (for possible removal) of path separators in both ends for every two parts concat'ed together, when I know for sure if the "leftmost" part has a path separator or not, and I am appending n number of simple path segments after it. Would be better if PathUtilites had some flexibility around such cases, will look into it.

**Comment by albertony** [comment]
Thank you for your review! I look into all your comments. It might take a little while, though, because I'm looking into some more similar cases etc.. Made it a Work-In-Progree PR, so that I get it through the build system etc.

File: src/lib/net/SecureSocket.cpp (hunk 1 above)
Comment (p12tic) [reply src/lib/net/SecureSocket.cpp:704]: The chance of bugs outweights the very small performance overhead. If this issue was fixed I could merge the PR.

**Pull request closed without merging**